syntax = "proto3";

package cms;

// The CMS gRPC service
service CMS {
  rpc export_document(ExportItemRequest) returns (stream ExportItemResponse);
  rpc create_document(CreateItemRequest) returns (CreateItemResponse); 
}

// Request and response messages
message ExportItemRequest {
  string author = 1;  
  repeated string tags = 2;
}

message ExportItemResponse {
  repeated Document document = 1;
}

message CreateItemRequest {
  Document document = 1;
}

message CreateItemResponse {
  Document document = 1;
} 

message Document {
  string id = 1;
  string title = 2;
  string content = 3;
  string author = 4;
  repeated string tags = 5;
}